src/pid.cpp

```diff
@@ -194,29 +194,21 @@ double Pid::compute_command(double error, double error_dot, const double & dt_s)
   // Calculate proportional contribution to command
   p_term = gains.p_gain_ * p_error_;
 
-  // Calculate the integral of the position error
-  i_error_ += dt_s * p_error_;
-
-  if (gains.antiwindup_ && gains.i_gain_ != 0) {
-    // Prevent i_error_ from climbing higher than permitted by i_max_/i_min_
-    std::pair<double, double> bounds =
-      std::minmax<double>(gains.i_min_ / gains.i_gain_, gains.i_max_ / gains.i_gain_);
-    i_error_ = std::clamp(i_error_, bounds.first, bounds.second);
-  }
-
   // Calculate integral contribution to command
-  i_term = gains.i_gain_ * i_error_;
-
-  if (!gains.antiwindup_) {
-    // Limit i_term so that the limit is meaningful in the output
-    i_term = std::clamp(i_term, gains.i_min_, gains.i_max_);
+  if (gains.antiwindup_) {
+    // Prevent i_term_ from climbing higher than permitted by i_max_/i_min_
+    i_term_ = std::clamp(i_term_ + gains.i_gain_ * dt_s * p_error_,
+      gains.i_min_, gains.i_max_);
+  } else {
+    i_term_ += gains.i_gain_ * dt_s * p_error_;
   }
 
   // Calculate derivative contribution to command
   d_term = gains.d_gain_ * d_error_;
 
   // Compute the command
-  cmd_ = p_term + i_term + d_term;
+  // Limit i_term so that the limit is meaningful in the output
+  cmd_ = p_term +  std::clamp(i_term_, gains.i_min_, gains.i_max_) + d_term;
```

Comment:
Not sure why i_term is being clamped again here when it was clamped above when antiwindup is true. Shouldn't it only be clamped when antiwindup is true

Reply:
I left that to keep the current behavior, it was clamped even with antiwindup being true.
from the docstring:
```
antiwindup Antiwindup functionality. When set to true, limits
        the integral error to prevent windup; otherwise, constrains the
        integral contribution to the control output. i_max and
        i_min are applied in both scenarios.
```

this was already questioned in #276 and I didn't want to change this behavior in this PR.